pgtrigger/core.py

```diff
@@ -437,13 +437,29 @@ def get_uri(self, model):
 
     def register(self, *models):
         """Register model classes with the trigger"""
+        # Compute the unique trigger function names that are already
+        # registered in order to prevent an accidental collision
+        registered_function_names = {
+            trigger.get_pgid(model) for model, trigger in registry.values()
+        }
+
         for model in models:
             uri = self.get_uri(model)
             if uri in registry:  # pragma: no cover
                 raise ValueError(
                     f'Trigger with name "{self.name}" is already'
-                    f' registerd for model "{model}"'
+                    f' registered for model "{model}"'
+                )
+
+            if (
+                self.get_pgid(model) in registered_function_names
+            ):  # pragma: no cover
+                raise ValueError(
+                    f'Trigger with name "{self.name}" on model "{model}"'
+                    ' has an trigger function name that is already taken.'
+                    ' Use a different name for the trigger.'
                 )
```

Comment:
There was no easy way to test this, and this error should never happen in practice. But it is still here as a safeguard in the case it does happen